src/csharp/Tensor.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public enum ElementType : long
+    {
+        undefined,
+        float32,
+        uint8,
+        int8,
+        uint16,
+        int16,
+        int32,
+        int64,
+        string_t,
+        bool_t,
+        float16,
+        float64,
+        uint32,
+        uint64,
+    }
+
+    public class Tensor : IDisposable
+    {
+        private IntPtr _tensorHandle;
+        private bool _disposed = false;
+
+        public Tensor(IntPtr data, Int64[] shape, ElementType type)
+        {
+            Result.VerifySuccess(NativeMethods.OgaCreateTensorFromBuffer(data, shape, (UIntPtr) shape.Length, type, out _tensorHandle));
+        }
+        internal IntPtr Handle { get { return _tensorHandle; } }
+
+        ~Tensor()
+        {
+            Dispose(false);
+        }
+
+        Int64[] GetShape()
```

Comment:
Shape, but no data?

Reply:
Good point, easy enough to add. We just didn't have a need for them yet so I implemented one as a test.

Reply:
Oops, wasn't even public. The unit test caught that.

Reply:
While you're at it, could we also call this function Shape() and Data() and so on?

Reply:
Yep! Was locally changed, now it's pushed.